lua/roslyn/slnutils.lua

```diff
@@ -26,48 +45,31 @@ function M.get_project_files(buffer)
     }
 end
 
---TODO: Lot of duplication and not good for method that recursively search the filesystem
----Tries to get csproj files recursively from the current directory.
----If we open a file that is not a child of the current directory, then
----we return `nil` as we haven't found the project files for that.
----Fallback to using normal behaviour to check solution and csproj files like before
----@param bufnr number
----@return RoslynNvimDirectoryWithFiles?
-function M.try_get_csproj_files(bufnr)
-    local found = false
-    local bufname = vim.api.nvim_buf_get_name(bufnr)
-
-    local files = vim.fs.find(function(name, path)
-        if bufname == vim.fs.joinpath(path, name) then
-            found = true
-        end
-
-        return name:match("%.csproj$") or name:match("%.sln$")
-    end, { type = "file", limit = math.huge })
-
-    local csproj_files = vim.iter(files)
-        :filter(function(it)
-            return it:match("%.csproj$")
-        end)
-        :totable()
-
-    if not found or #files ~= #csproj_files or #files < 1 then
-        return nil
+--- Attempts to find `.csproj` files in the current working directory (CWD).
+--- This function searches recursively through the files in the CWD.
+--- If a `.csproj` file is found, it returns the directory path and a list of matching files.
+--- If no `.csproj` files are found or the file is outside the CWD, `nil` is returned.
+--- Falls back to normal behavior for checking solution and project files if no match is found.
+---
+--- @return RoslynNvimDirectoryWithFiles? A table containing the directory path and a list of found `.csproj` files, or `nil` if none are found.
+function M.try_get_csproj_files()
+    local cwd = assert(vim.uv.cwd())
+
+    local csprojs = find_files_with_extension(cwd, ".csproj")
```

Comment:
Aaah, I think we also need to check that there isn't any solution files in the `cwd`. At least in one a project I have on my machine, there is actually both a csproj file and a solution file in the same folder.

I think it could be okay to just do something like this:

```cs
local sln = find_files_with_extension(cwd, ".sln")

if #csprojs > 0 and #sln < 1 then
```

Even though we are search the cwd twice, I don't think it should impact performance🤔 If you want, you could also change it so that we only search cwd once, but then I don't think we could reuse the function in a good way.

It is up to you😄 I am fine with either

Reply:
Thanks, I totally forgot about this behaviour. 😅

Reply:
Yeah me too😅 I was just about to merge it blindly since it's pretty easy to reason about the code, but I decided I could test it just to be sure, and then immediately got hit by that haha. However, now I think it is good!

I will probably add something to support some kind of broad search or another way of selecting a solution file that is not in the root or something, but I will do that in a follow up sometime